src/components/Governance/VpTable.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useEffect, useState } from 'react';
+import { useRecoilValue } from 'recoil';
+import { userVpState } from '../../context/userVpState';
+import CustomTable from '../DataDisplay/CustomTable';
+import { Box } from '@chakra-ui/react';
+import CosmosImg from '../../assets/chains/cosmos.png';
+import NeutronImg from '../../assets/chains/neutron.jpg';
+import StrideImg from '../../assets/chains/stride.png';
+import CustomSkeleton from '../skeleton/CustomSkeleton';
+import { shortenCosmosAddress } from '../../utils/common';
+
+const chainLogoImg = {
+    Cosmos: CosmosImg,
+    Neutron: NeutronImg,
+    Stride: StrideImg
+};
+
+
+const VpTable = ({ sizeOfAllProposals }: { sizeOfAllProposals: object }) => {
+    const userVp: any = useRecoilValue(userVpState);
+    const [tableArray, setTableArray] = useState<any[]>([]);
+
+
+    useEffect(() => {
+        if (!userVp || Object.keys(sizeOfAllProposals).length === 0) return;
```

Comment:
Instead of putting this code directly inside of `useEffect`, transfer this code into a local function and call that function inside of `useEffect`.